src/CRtspSession.cpp

```diff
@@ -21,281 +26,407 @@ CRtspSession::CRtspSession(WiFiClient& aClient, CStreamer * aStreamer) : LinkedL
-                    parseSucceeded = true;
+                if ( p[ 2 ] != ' ' && p[ 2 ] != '\t' ) // no space. ending search
                     break;
-                }
-                m_CSeq[n] = c;
+
+                // clearing and looking for another continuation
+                *p = ' ';
+                ++p;
+                *p = ' ';
             }
-            m_CSeq[n] = '\0';
-            break;
         }
-    }
-    if (!parseSucceeded) return false;
 
-    // Also: Look for "Content-Length:" (optional)
-    for (j = i; (int)j < (int)(CurRequestSize-15); ++j)
-    {
-        if (CurRequest[j]    == 'C'  && CurRequest[j+1]  == 'o'  &&
-            CurRequest[j+2]  == 'n'  && CurRequest[j+3]  == 't'  &&
-            CurRequest[j+4]  == 'e'  && CurRequest[j+5]  == 'n'  &&
-            CurRequest[j+6]  == 't'  && CurRequest[j+7]  == '-'  &&
-            (CurRequest[j+8] == 'L' || CurRequest[j+8]   == 'l') &&
-            CurRequest[j+9]  == 'e'  && CurRequest[j+10] == 'n' &&
-            CurRequest[j+11] == 'g' && CurRequest[j+12]  == 't' &&
-            CurRequest[j+13] == 'h' && CurRequest[j+14] == ':')
+        // transport settings: proto, ports, etc
+        if ( m_RtspCmdType == RTSP_SETUP && 0 == strncmp( "Transport:", cur_pos, 10 ) )
         {
-            j += 15;
-            while (j < CurRequestSize && (CurRequest[j] ==  ' ' || CurRequest[j] == '\t')) ++j;
-            unsigned num;
-            if (sscanf(&CurRequest[j], "%u", &num) == 1) m_ContentLength = num;
-        }
-    }
+            cur_pos += 10;
+            while( *cur_pos && isspace( *cur_pos ) )
+                ++cur_pos;
+            
+            if ( 0 != strncmp( cur_pos, "RTP/AVP", 7) ) // std says this is mandatory part
+                return false;
+            
+            cur_pos += 7;
+
+            if ( 0 == strncmp( cur_pos, "/TCP", 4 ) ) // TCP is also good?
+            {
+                m_TcpTransport = true;
+                cur_pos += 4;
+            }
+            else
+                m_TcpTransport = false;
+
+            if ( debug ) printf( "+ Transport is %s\n", (m_TcpTransport ? "TCP" : "UDP") );
+
+            m_ClientRTPPort = -1;
+
+            // now looking for sub-params like clent_port=
+            char *next_part, last_char;
+            for(;;)
+            {
+                while( *cur_pos == ';' || *cur_pos == ' ' || *cur_pos == '\t' )
+                    ++cur_pos;
+
+                if ( ! *cur_pos )
+                    return false;
+
+                if ( *cur_pos == '\r' && cur_pos[ 1 ] == '\n' )
+                    break;
+
+                next_part = strpbrk( cur_pos, ";\r" ); // gettin pointer to the next sub-param if any
+                if( ! next_part )
+                    return false;
+
+                last_char = *next_part; // in case we'll need to put \0 here
+
+                if ( m_ClientRTPPort == -1 && 0 == strncmp( cur_pos, "client_port=", 12 ) ) // "client_port" "=" port [ "-" port ]

```

Comment:
a comparison to -1 and an unsigned value will always be false, check for zero instead